src/gui/pages_menu/KM_GUIMenuLobby.pas

```diff
@@ -1133,7 +1143,18 @@ function TKMMenuLobby.SlotsAvailable: Byte;
 end;
 
 
-function TKMMenuLobby.DropColPlayersCellClick(Sender: TObject; const X, Y: Integer): Boolean;
+procedure TKMMenuLobby.DropBoxPlayers_Show(Sender: TObject);
+var DropBox: TKMDropColumns;
+begin
+  if Sender is TKMDropColumns then
+  begin
+    DropBox := TKMDropColumns(Sender);
+    fDropBoxPlayerSlot_ItemIndexes[DropBox.Tag] := DropBox.ItemIndex;
```

Comment:
This seems kind of fragile, if we forget to update fDropBoxPlayerSlot_ItemIndexes somewhere. For example I'm not sure your solution will correctly handle when you select a save and remaining slots are set to closed/AIs automatically. Or when players join/leave the lobby and item indexes are updated.

Why can't we handle this problem completely in DropColPlayersCellClick? That would be much simpler than first applying the wrong settings and then undoing mistakes by keeping track of what the values were before we set them wrong?

I don't think we care about the case where the user already has the correct number of players, and sets an extra AI to "all" at the end. That's the user being silly and should happen very rarely. In that case it would be fine to let that slot be set to AI (but leave ones below it) and leave the user with one extra AI. Or set that slot to closed or open instead. Doesn't really matter because it should be very rare.